### PR TITLE
[Speech2Text2] Enable tokenizers

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Tokenizer files have been added here:
- https://huggingface.co/facebook/s2t-wav2vec2-large-en-tr/commit/ae0ccd057a5c698ddb7fd439c9238ae49b8865d8
for all s2t2 models: https://huggingface.co/models?other=speech2text2

This PR is the follow-up PR of this one: https://github.com/huggingface/transformers/pull/13186

It is made sure that full backwards compatibility is kept which means that a tokenizer can still be instantiated without a merges file for decoding only.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/speech_to_text_2/tokenization_speech_to_text_2.py [hunk 1]
```diff
@@ -73,28 +93,123 @@ class Speech2Text2Tokenizer(PreTrainedTokenizer):
...
         )
 
+        self.do_lower_case = do_lower_case
+
         with open(vocab_file, encoding="utf-8") as vocab_handle:
             self.encoder = json.load(vocab_handle)
         self.decoder = {v: k for k, v in self.encoder.items()}
 
+        if merges_file is None:
+            logger.info(f"No merges files provided. {self.__class__.__name__} " "can only be used for decoding.")
```
Comment (sgugger) [inline src/transformers/models/speech_to_text_2/tokenization_speech_to_text_2.py:123]:
```suggestion
            logger.info(f"No merges files provided. {self.__class__.__name__} can only be used for decoding.")
```

File: src/transformers/models/speech_to_text_2/tokenization_speech_to_text_2.py [hunk 2]
```diff
@@ -73,28 +93,123 @@ class Speech2Text2Tokenizer(PreTrainedTokenizer):
...
+
+    def _tokenize(self, text):
+        """Tokenize a string."""
+
+        if self.bpe_ranks is None:
+            raise ValueError(
+                "This tokenizer was instantiated without a `merges.txt` file, so"
+                " that it can only be used for decoding, not for encoding."
+                "Make sure to provide `merges.txt` file at instantiation to enable "
+                "enocding."
```
Comment (sgugger) [inline src/transformers/models/speech_to_text_2/tokenization_speech_to_text_2.py:199]:
```suggestion
                "encoding."
```

**Review by sgugger — APPROVED** [review]
Thanks for adding those!

**Review by anton-l — APPROVED** [review]
Looks good, thanks for porting them!

**Pull request merged**